core/indexer/elasticsearchDatabaseWritter.go

```diff
@@ -0,0 +1,110 @@
+package indexer
+
+import (
+	"bytes"
+	"context"
+	"io"
+	"net/http"
+
+	"github.com/elastic/go-elasticsearch/v7"
+	"github.com/elastic/go-elasticsearch/v7/esapi"
+)
+
+type databaseWriter struct {
+	client *elasticsearch.Client
+}
+
+func newDatabaseWriter(cfg elasticsearch.Config) (*databaseWriter, error) {
+	es, err := elasticsearch.NewClient(cfg)
+	if err != nil {
+		return nil, err
+	}
+
+	return &databaseWriter{client: es}, nil
+}
+
+// CheckAndCreateIndex will check if a index exits and if dont will create a new one
+func (dw *databaseWriter) checkAndCreateIndex(index string, body io.Reader) error {
+	res, err := dw.client.Indices.Exists([]string{index})
+	if err != nil {
+		return err
+	}
+
+	defer closeESResponseBody(res)
```

Comment:
maybe move the function from common.go inside this file.

Reply:
done.